drivers/usbmisc/fusb302.c

```diff
@@ -0,0 +1,1941 @@
+/****************************************************************************
+ * drivers/usbmisc/fusb302.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <assert.h>
+#include <errno.h>
+#include <poll.h>
+#include <debug.h>
+
+#include <nuttx/compiler.h>
+#include <nuttx/fs/fs.h>
+#include <nuttx/kmalloc.h>
+#include <nuttx/kthread.h>
+#include <nuttx/mutex.h>
+#include <nuttx/wqueue.h>
+#include <nuttx/i2c/i2c_master.h>
+
+#include <nuttx/usb/fusb302.h>
+#include <sys/ioctl.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#ifndef FUSB302_I2C_FREQUENCY
+#  define FUSB302_I2C_FREQUENCY 800000
```

Comment:
It this correct? Normally I2C is 100KHz (Standard Mode), 400KHz (Full Speed Mode), 1MHz (Fast-mode Plus / Fm+) or 3.2MHz (High Speed Mode)

Reply:
@acassis Fast mode plus at 1MHz. I just don't like running things at at absolute maximum, but I will make 1MHz  the default.